### PR TITLE
expression: support customizing FieldTypes of arguments for benchmark framework

### PR DESCRIPTION
**Pull request by SunRunAway**
### What problem does this PR solve? 

support customizing FieldTypes of arguments for benchmark framework.
previous PR of #12224

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

 - Increased code complexity

Related changes

 - None

Release note

 - None

### CODE REVIEW

File: expression/bench_test.go [hunk 1]
```diff
@@ -436,31 +454,50 @@ func testVectorizedEvalOneVec(c *C, vecExprCases vecExprBenchCases) {
 			switch testCase.retEvalType {
 			case types.ETInt:
 				for i := 0; i < input.NumRows(); i++ {
+					c.Assert(c1.IsNull(i), Equals, c2.IsNull(i), commentf(i))
 					c.Assert(c1.IsNull(i) != c2.IsNull(i) || (!c1.IsNull(i) && c1.GetInt64(i) != c2.GetInt64(i)), IsFalse)
```
Comment (Reminiscent) [inline expression/bench_test.go]: Should we delete this line?

Comment (Reminiscent) [inline expression/bench_test.go]: Should we modify this line?

Comment (SunRunAway) [reply expression/bench_test.go]: updated.

**Pull request merged**